dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/exchange/Response.java

```diff
@@ -173,4 +183,55 @@ public String toString() {
         return "Response [id=" + mId + ", version=" + mVersion + ", status=" + mStatus + ", event=" + mEvent
                 + ", error=" + mErrorMsg + ", result=" + (mResult == this ? "this" : mResult) + "]";
     }
+
+    public static class AppResult {
+
+        protected Map<String, String> attachments = new HashMap<String, String>();
+
+        protected Object result;
+
+        protected Throwable exception;
+
+        public Map<String, String> getAttachments() {
+            return attachments;
+        }
+
+        public void setAttachments(Map<String, String> attachments) {
+            this.attachments = attachments;
+        }
+
+        public Object getResult() {
+            return result;
+        }
+
+        public void setResult(Object result) {
+            this.result = result;
+        }
+
+        public Throwable getException() {
+            return exception;
+        }
+
+        public void setException(Throwable exception) {
+            this.exception = exception;
+        }
+    }
+
+    public static class DecodeableAppResult implements Codec, Decodeable {

```

Comment:
I searched this class usage, but found it is useless.

Reply:
removed.